backend-tests/tests/test_deployments.py

```diff
@@ -0,0 +1,257 @@
+import pytest
+from datetime import datetime, timedelta
+import time
+import os
+import requests
+
+import api.client
+from api.client import ApiClient
+import api.deviceauth as deviceauth_v1
+import api.useradm as useradm
+import api.inventory as inventory
+from common import User, Device, Authset, Tenant, \
+        create_tenant, create_tenant_user, \
+        make_accepted_devices, clean_mongo, mongo
+
+
+def upload_image(filename, auth_token, description="abc"):
+    image_path_url = api.client.GATEWAY_URL + "/api/management/v1/deployments/artifacts"
+    r = requests.post(
+        image_path_url,
+        verify=False,
+        headers={"Authorization": "Bearer " + auth_token},
+        files=(("description", (None, description)),
+               ("size", (None, str(os.path.getsize(filename)))),
+               ("artifact", (filename, open(filename, 'rb'),
+                             "application/octet-stream"))))
+    assert r.status_code == 201
+
+
+@pytest.fixture(scope='function')
+def setup_deployments_enterprise_test(clean_mongo, nr_devices=100):
+    tenant = create_tenant('acme')
```

Comment:
cool, but it makes sense to test with at least 2 tenants.

this is to make sure we're not adding/screwing up data for the 'other' tenant.

I know it looks like it can't happen, but it can - say if the automatic database selection inside the storage layer is screwed up - and points to the single default db. then, 1 tenant will have all the symptoms of a passing test. for 2 tenants it will be immediately obvious that some data changed for both of them, when it shouldn't.